caproto/_data.py

```diff
@@ -146,7 +146,8 @@ def string_encoding(self):
 
 
 class ChannelData:
-    data_type = ChType.LONG
+    # subclass must define a `data_type` class attribute, set to a value of the
+    # ChType enum
 
     def __init__(self, *, value=None, timestamp=None, status=0, severity=0,
                  string_encoding='latin-1', alarm_status=None,
```

Comment:
Notes to self:
* `values` should be called `data` for consistency with everything else
* `string_encoding` should probably only be a kwarg for enum, str, and char channels
* `alarm_status` makes sense (as notes in docstring, it's for sharing one alarm between channels)
* `timestamp`, `status`, `severity` should be lumped into `metadata`, same as in the `set_dbr_data` method
* If alarm-related metadata is provided (i.e., status, severity) it should get applied to the alarm status instance, not silently ignored.